adam-core/src/main/scala/org/bdgenomics/adam/models/ReferenceRegion.scala

```diff
@@ -510,12 +560,30 @@ case class ReferenceRegion(
   }
 
   override def hashCode: Int = {
-    var result = 37
-    result = 41 * result + (if (referenceName != null) referenceName.hashCode else 0)
-    result = 41 * result + start.hashCode
-    result = 41 * result + end.hashCode
-    result = 41 * result + (if (strand != null) strand.ordinal() else 0)
-    result
+    // add 37 here because this is the start of the hashCode
+    val nameHashCode = 37 + {
```

Comment:
https://github.com/bigdatagenomics/adam/commit/c4b23e0be0898b1c662f5a08dd7fe9b8236f1775

It might have been a performance issue. @heuermh asked the same question in 2015. Interesting.